src/Base/Exception.h

```diff
@@ -196,812 +151,382 @@ class BaseExport Exception: public BaseClass
     TYPESYSTEM_HEADER_WITH_OVERRIDE();
 
 public:
+    explicit Exception(std::string message = "FreeCAD Exception");
     ~Exception() noexcept override = default;
 
     Exception& operator=(const Exception& inst);
     Exception& operator=(Exception&& inst) noexcept;
 
     virtual const char* what() const noexcept;
+    virtual void reportException() const;  // once only
 
-    /// Reports exception. It includes a mechanism to only report an exception once.
-    virtual void ReportException() const;
-
-    inline void setMessage(const char* sMessage);
-    inline void setMessage(const std::string& sMessage);
-    // what may differ from the message given by the user in
-    // derived classes
-    inline std::string getMessage() const;
+    inline void setMessage(const std::string& message);
+    inline std::string getMessage() const;  // what may differ from in derived class

```

Comment:
This comment broke someway along the way.
Feel free to use the GH ui to just apply suggestion as a commit if you like.
```suggestion
    // what may differ from the message given by the user in
    // derived classes
    inline std::string getMessage() const;
```